addon/mixins/route.js

```diff
@@ -123,14 +123,17 @@ const RouteMixin = Mixin.create({
     }
 
     const currentPage = options.startingPage === undefined ? 0 : options.startingPage-1;
-    const perPage         = options.perPage || 25;
-    const perPageParam    = options.perPageParam || 'per_page';
-    const pageParam    = options.pageParam || 'page';
+    const perPage = options.perPage || 25;
+
+    // check if user passed in param w/ infinityModel, else check if defined on the route (for backwards compat), else default
+    const perPageParam = typeOfCheck(options.perPageParam, get(this, 'perPageParam'), 'per_page');
```

Comment:
Somehow we need to allow the user to pass null in the infinityRoute, but if they don't fallback to route, then fallback to default.  

So if pass `null`, don't send up with request.  Else check if route defines `null`.  Else always send up those params with a request.